### PR TITLE
Stats safer encode

### PR DESCRIPTION
**Pull request by max-hoffman**
We previously used commas as serialization boundaries for multi-field stats tuples (bucket bounds). That worked well for numeric values, and doesn't work well for strings with commas. This uses the prolly serialization code to more safely round trip tuples.

We still use commas to separate MCV counts, which are integers, and newlines (`\n`) for index types. If types can have newlines at some point we would want to switch that to prolly encoding as well.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>9746326</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>9746326</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>6ce6230</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>6ce6230</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>4a7e8ac</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>4a7e8ac</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/statsnoms/iter.go [hunk 1]
```diff
@@ -159,7 +159,11 @@ func (s *statsIter) Next(ctx *sql.Context) (sql.Row, error) {
 
 func (s *statsIter) ParseRow(rowStr string) (sql.Row, error) {
 	var row sql.Row
-	for i, v := range strings.Split(rowStr, ",") {
+	parts := strings.Split(rowStr, ",")
+	if len(parts) > len(s.currentTypes) {
```
Comment (nicktobey) [inline go/libraries/doltcore/sqle/statsnoms/iter.go:163]: So I see how this prevents a panic. But is it correct? I imagine we want to parse as much as we can and return 0 values for the remaining columns, right?

**Review by nicktobey — APPROVED** [review]

File: go/libraries/doltcore/sqle/statsnoms/iter.go (hunk 1 above)
Comment (max-hoffman) [reply go/libraries/doltcore/sqle/statsnoms/iter.go:163]: Ah i meant to revert, I was using to before to see if the database had any other loading issues. Thanks for pointing out

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>8437aa2</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>8437aa2</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**